src/background/exceptions.js

```diff
@@ -142,10 +142,20 @@ async function updateFilters() {
     removeRuleIds,
   });
 
-  console.info('[exceptions] DNR rules for filters updated successfully');
+  console.info('[exceptions] Updated DNR rules');
 }
 
 if (__PLATFORM__ === 'chromium' || __PLATFORM__ === 'safari') {
   // Update exceptions filters every time TrackerDB updates
-  engines.addChangeListener(engines.TRACKERDB_ENGINE, updateFilters);
+  // It happens when all engines are updated
+  OptionsObserver.addListener(
+    'filtersUpdatedAt',
+    async function updateExceptions(value, lastValue) {
+      // Only update exceptions filters if the value has changed and is set to timestamp.
+      // It will happen only after successful update of the engines.
+      if (lastValue !== undefined && value !== 0) {
+        await updateFilters();
```

Comment:
this was previously happening for trackerdb updates only. 
IDK if it this better or worse now. Do we want to run it when Custom Filters change?

Reply:
Yes, technically, it was more accurate. However, it only applies to exceptions, and most users will have none, or very few. The case is to stop keeping complex logic just for one place. Even though the change is not accurate, I think it is worth simplifying the structure and re-use patterns as much as possible.

I added a condition for only touching DNR rules if there is a change - in the current implementation triggering an update for users without exceptions does nothing.